File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    idx = 0
    box = 1
    stack = []
    while box <= len(order):
        if box == order[idx]:
            idx += 1
        else:
            stack.append(box);
        box += 1
        
        while stack and stack[-1] == order[idx]:
            stack.pop()
            idx += 1
    return idx
        

File: 프로그래머스/2/131704. 택배상자/택배상자.js
function solution(order) {
    let stack = [];
    let idx = 0;
    let box = 1;
    
    while(box <= order.length){
        if(box === order[idx]){
            idx++;
        } else {
            stack.push(box);
        }
        box++;
        
        while(stack.length > 0 && stack[stack.length - 1] === order[idx]){
            stack.pop();
            idx++;
        }
    }
    return idx;
}